### PR TITLE
(maint) Some minor improvements to garbage generation

### PR DESCRIPTION
**Pull request by pcarlisle**

### CODE REVIEW

File: spec/unit/util/tagging_spec.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #! /usr/bin/env ruby
+# coding: utf-8
```
Comment (Magisus) [inline spec/unit/util/tagging_spec.rb]: Was this added intentionally?

**Comment by Magisus** [comment]
Out of curiosity, how did you choose which strings to freeze?

File: lib/puppet/util/tagging.rb [hunk 2]
```diff
@@ -78,16 +78,14 @@ def set_tags(tag_source)
     @tags = tag_source.tags
   end
 
-  # Return a copy of the tag list, so someone can't ask for our tags
-  # and then modify them.
   def tags
     @tags ||= new_tags
-    @tags.dup
+    @tags
```
Comment (pcarlisle) [inline lib/puppet/util/tagging.rb]: definitely need a second opinion on the safety of this change

**Comment by pcarlisle** [comment]
using memory_profile benchmarks it prints a summary of the most allocated strings

File: spec/unit/util/tagging_spec.rb (hunk 1 above)
Comment (pcarlisle) [reply spec/unit/util/tagging_spec.rb]: my editor did it automatically because it detected something non-ascii in this file

**Comment by joshcooper** [comment]
jenkins please test this

File: lib/puppet/util/tagging.rb (hunk 2 above)
Comment (joshcooper) [reply lib/puppet/util/tagging.rb]: This makes me a little nervous, but acceptance tests pass. What do you think?

**Comment by kris-bosland** [comment]
Hi @pcarlisle, with the 5.5.x branch being cut, please retarget that branch if appropriate.  Thanks!

**Pull request merged**